### PR TITLE
Run tests in parallel and coverage diff report to PR comments

### PR DESCRIPTION
**Pull request by liudmylaru**
## Description

This PR is aiming 3 goals:
- make unittests running in parallel (using pytest-xdist) library, which speeds up tests by 30%
- make useful coverage message for open PR with coverage difference (percentage of new or modified lines that are covered by tests)
- calculate overall coverage and addt his information int oPR comment

Other changes - small enhancements - removing duplicated code from test_model_framework_tensorflow_model.py, reordering list of libraries in requirements.txt files, cleaning coveragerc.

Jira: https://jira.oci.oraclecorp.com/browse/ODSC-41181

## What was done

- added final job in Actions to combine and produce cov diff report
- updated test-requirements.txt and dev-requirements.txt
- removed duplicated code in test_model_framework_tensorflow_model.py
- cleaned .coveragerc
- added pragma: no cover to Exception classes
- added paths to omit in report for deprecated code to .coveragerc

### CODE REVIEW

File: tests/unitary/with_extras/model/test_model_framework_tensorflow_model.py [hunk 1]
```diff
@@ -63,11 +62,9 @@ class TestTensorFlowModel:
     """Unittests for the TensorFlowModel class."""
 
     def setup_class(cls):
```
[the remark this answers is not in the document]
Comment (mrDzurb) [reply tests/unitary/with_extras/model/test_model_framework_tensorflow_model.py:64]: I usually add ```@classmethod ``` on top, but looks like this is not mandatory for the pytest ```setup_class```.

**Review by mrDzurb — DISMISSED** [review]

File: tests/unitary/with_extras/model/test_model_framework_tensorflow_model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (liudmylaru) [reply tests/unitary/with_extras/model/test_model_framework_tensorflow_model.py:64]: QQ pointed to this documentation - https://docs.pytest.org/en/6.2.x/xunit_setup.html#class-level-setup-teardown. I will add @classmethod.

File: tests/unitary/with_extras/model/test_model_framework_tensorflow_model.py
[the remark this answers is not in the document]
Comment (liudmylaru) [reply tests/unitary/with_extras/model/test_model_framework_tensorflow_model.py:1]: added `--dist loadfile`

File: tests/unitary/with_extras/model/test_model_framework_tensorflow_model.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (liudmylaru) [reply tests/unitary/with_extras/model/test_model_framework_tensorflow_model.py:65]: I've made change locally and pushed it PR

**Review by qiuosier — APPROVED** [review]

**Review by qiuosier — APPROVED** [review]

**Review by mayoor — APPROVED** [review]

**Pull request merged**